src/route/dataset-route.ts

```diff
@@ -174,13 +201,47 @@ apiRoute.get('/:dataset/xlsx', async (req, res) => {
     });
 });
 
+apiRoute.get('/:dataset/preview', async (req, res) => {
+    const datasetID = req.params.dataset;
+    if (datasetID === undefined || datasetID === null) {
+        res.status(404);
+        res.json({ message: 'Dataset not found... You must specify a dataset ID' });
+        return;
+    }
+    if (isValidUUID(datasetID) === false) {
+        res.status(404);
+        res.json({ message: 'Dataset not found...File ID is not Valid.' });
+        return;
+    }
+    const dataset = await Dataset.findOneBy({ id: datasetID });
+    if (dataset === undefined || dataset === null) {
+        res.status(404);
+        res.json({ message: 'Dataset not found... Dataset ID not found in Database' });
+        return;
+    }
+    const page_number_str: string = req.query.page_number || req.body?.page_number;
+    const page_size_str: string = req.query.page_size || req.body?.page_size;
+    const page_number: number = Number.parseInt(page_number_str, 10) || 1;
+    const page_size: number = Number.parseInt(page_size_str, 10) || DEFAULT_PAGE_SIZE;
+    const processedCSV = await processCSVFromBlobStorage(dataset, page_number, page_size);
+    if (!processedCSV.success) {
+        res.status(400);
+    }
+    res.json(processedCSV);
+});
+
 apiRoute.get('/:dataset/view', async (req, res) => {
     const datasetID = req.params.dataset;
     if (datasetID === undefined || datasetID === null) {
         res.status(404);
```

Comment:
There's another Response code for Bad Request.

Reply:
Moved out to another function which standardised the response to `400`